### PR TITLE
Introducing form component

### PR DESCRIPTION
**Pull request by mlunoe**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/Util/Util.js [hunk 1]
```diff
@@ -121,6 +121,43 @@ function exclude(object, props) {
...
+ * @static
+ * @category Array
+ * @param {Array} array The array to flatten.
+ * @returns {Array} Returns the new flattened array.
+ * @example
+ *
+ * _.flatten([1, [2, 3, [4]]]);
+ * // => [1, 2, 3, [4]]
+ */
+function flatten(array) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Introducing `flatten`

File: src/Util/Util.js [hunk 2]
```diff
@@ -121,6 +121,43 @@ function exclude(object, props) {
...
+ * @returns {boolean} Returns `true` if `value` is correctly classified, else `false`.
+ * @example
+ *
+ * _.isArguments(function() { return arguments; }());
+ * // => true
+ *
+ * _.isArguments([1, 2, 3]);
+ * // => false
+ */
+function isArguments(value) {
```
Comment (mlunoe) [inline src/Util/Util.js]: This shows weird. I just reordered some functions alphabetically

File: src/Util/Util.js [hunk 3]
```diff
@@ -265,6 +282,145 @@ function values(object) {
   return object ? baseValues(object, Object.keys(object)) : [];
 }
 
+// Functions from Underscore 1.9.0
+
+ // Internal recursive comparison function for `isEqual`.
+function eq(a, b, aStack, bStack) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Introducing `isEqual`

File: src/Util/Util.js [hunk 4]
```diff
@@ -442,20 +598,34 @@ const Util = {
     return a.length !== b.length;
   },
 
+  capitalize: function (string) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Introducing `capitalize` and `isArray`

File: src/Util/Util.js [hunk 5]
```diff
@@ -280,52 +436,6 @@ function clone(object) {
...
- * @param {Function} func A callback function to be called
- * @param {Number} wait How long to wait
- * @param {Boolean} immediate If it should be called immediately
- * @returns {Function} A function, that, as long as it continues to be
- * invoked, will not be triggered. The function will be called
- * after it stops being called for N milliseconds.
- * If `immediate` is passed, trigger the function on the leading edge,
- * instead of the trailing.
- */
-function debounce(func, wait, immediate) {
```
Comment (mlunoe) [inline src/Util/Util.js]: Debounce was not used anywhere

File: src/Util/Util.js [hunk 6]
```diff
@@ -465,20 +575,33 @@ const Util = {
...
+
+    return string.charAt(0).toUpperCase() + string.slice(1, string.length);
+  },
+
+  isArray: function (arg) {
+    return Object.prototype.toString.call(arg) === '[object Array]';
+  },
+
   // Add external lodash functions
-  noop: noop,
```
Comment (mlunoe) [inline src/Util/Util.js]: Again, alphabetizing

**Comment by rcorral** [comment]
Even @ashenden made it to the commit history!

File: src/Form/FieldCheckbox.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+import React from 'react';
+
+const METHODS_TO_BIND = ['handleChange'];
+
+export default class FieldCheckbox extends React.Component {
+  constructor() {
+    super();
+
+    METHODS_TO_BIND.forEach((method) => {
```
Comment (daiyiwhale) [inline src/Form/FieldCheckbox.js]: I think we can use `BindMixin` here (:

File: src/Form/FieldPassword.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+const DEFAULT_PASSWORD_TEXT = '••••••';
+
+const METHODS_TO_BIND = ['handleOnFocus'];
+
+export default class FieldPassword extends FieldInput {
+  constructor() {
+    super();
+
+    METHODS_TO_BIND.forEach((method) => {
```
Comment (daiyiwhale) [inline src/Form/FieldPassword.js]: `BindMixin` here as well

File: src/Form/Form.js [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+  constructor() {
+    super();
+
+    this.state = {
+      model: {},
+      editingField: '',
+      erroredFields: {}
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
```
Comment (daiyiwhale) [inline src/Form/Form.js]: Another `BindMixin` opportunity

File: src/Form/FieldCheckbox.js (hunk 7 above)
Comment (rcorral) [inline src/Form/FieldCheckbox.js]: +1

Comment (mlunoe) [inline src/Form/FieldCheckbox.js]: I'm going to change this one in my upcoming PR

Comment (mlunoe) [inline src/Form/FieldCheckbox.js]: Nevermind. I can make work :)

Comment (mlunoe) [inline src/Form/FieldCheckbox.js]: done

File: src/Form/FieldPassword.js (hunk 8 above)
Comment (mlunoe) [inline src/Form/FieldPassword.js]: done

File: src/Form/Form.js (hunk 9 above)
Comment (mlunoe) [inline src/Form/Form.js]: done

File: src/Form/FieldPassword.js (hunk 8 above)
Comment (mlunoe) [inline src/Form/FieldPassword.js]: Actually, I can't use it here, since I am extending from `FieldInput`

**Pull request merged**